src/plots/cartesian/constraints.js

```diff
@@ -53,14 +55,14 @@ exports.handleConstraintDefaults = function(containerIn, containerOut, coerce, a
     // 'matches' wins over 'scaleanchor' (for now)
     var scaleanchor, scaleOpts;
 
-    if(!matches && containerIn.scaleanchor && !(containerOut.fixedrange && constrain !== 'domain')) {
+    if(!matches && (containerIn.scaleanchor || opts.scaleanchorDflt) && !(containerOut.fixedrange && constrain !== 'domain')) {
         scaleOpts = getConstraintOpts(constraintGroups, thisID, allAxisIds, layoutOut, constrain);
         scaleanchor = Lib.coerce(containerIn, containerOut, {
             scaleanchor: {
                 valType: 'enumerated',
                 values: scaleOpts.linkableAxes || []
             }
-        }, 'scaleanchor');
+        }, 'scaleanchor', opts.scaleanchorDflt);
```

Comment:
To have a more clear interface with input arguments, would you mind setting
```js
var scaleanchorDflt = opts.scaleanchorDflt
```
on top of the function and then use all those optional attributes within the function?
It can also help reduce the chance of typos that may silently happen in the object properties. 
Thanks!

Reply:
Done in d71c3b345d2eecb7b3087c383f00f5f102b9c2f6